packages/handler-kit-azure-func/src/function.ts

```diff
@@ -13,38 +13,42 @@ import * as H from "@pagopa/handler-kit";
 import * as L from "@pagopa/logger";
 
 import { getLogger } from "./logger";
-import { getTriggerBindingData } from "./trigger";
 
-const hasLogger = <R, I>(
+const isHandlerEnvironment = <R, I>(
   u: unknown
-): u is R & { logger: L.Logger } & H.HandlerEnvironment<I> =>
-  typeof u === "object" && u !== null && "logger" in u;
+): u is R & H.HandlerEnvironment<I> =>
+  typeof u === "object" && u !== null && "logger" in u && "input" in u;
 
 // Populates Handler dependencies reading from azure.Context
-const azureFunctionTE = <I, A, R>(
-  h: H.Handler<I, A, R>,
-  deps: Omit<R, "logger"> & { inputDecoder: t.Decoder<unknown, I> }
-) =>
-  flow(
-    sequenceS(RE.Apply)({
-      logger: RE.fromReader(getLogger),
-      input: getTriggerBindingData(),
-    }),
-    TE.fromEither,
-    TE.map(({ input, logger }) => ({ input, logger, ...deps })),
-    TE.filterOrElse(hasLogger<R, I>, () => new Error("Unmeet dependencies")),
-    TE.chainW(h)
-  );
+const azureFunctionTE =
+  <I, A, R>(
+    h: H.Handler<I, A, R>,
+    deps: Omit<R, "logger" | "input"> & { inputDecoder: t.Decoder<unknown, I> }
+  ) =>
+  (messages: unknown, ctx: InvocationContext) =>
```

Comment:
I'll rename `messages` to a more explicit name, maybe `input` ?

I think that this function can accept only one parameter: `payload`, that has a composed type (`{ input: unknown, ctx: InvocationContext }`) - in this way it can be more pipeable